lightning/src/ln/channel.rs

```diff
@@ -53,19 +53,37 @@ enum InboundHTLCRemovalReason {
 }
 
 enum InboundHTLCState {
-	/// Added by remote, to be included in next local commitment tx.
+	/// Offered by remote, to be included in next local commitment tx. I.e., the remote sent an
+	// update_add_htlc message for this HTLC.
 	RemoteAnnounced(PendingHTLCStatus),
-	/// Included in a received commitment_signed message (implying we've revoke_and_ack'ed it), but
-	/// the remote side hasn't yet revoked their previous state, which we need them to do before we
-	/// accept this HTLC. Implies AwaitingRemoteRevoke.
-	/// We also have not yet included this HTLC in a commitment_signed message, and are waiting on
-	/// a remote revoke_and_ack on a previous state before we can do so.
+	/// Included in a received commitment_signed message (implying we've revoke_and_ack'd it), but
+	/// the remote hasn't yet revoked their previous state (see the example below). We have not yet
+	/// included this HTLC in a commitment_signed message because we are waiting on the remote's
+	/// aforementioned state revocation. The reason we need this remote RAA (revoke_and_ack) is
+	/// because every time we create a new "state", i.e. every time we create our next
+	/// per_commitment_point (see [BOLT #2]), we consume a prior revocation point (aka the
```

Comment:
Hmm, this reads strange to me, as we never "construct"  per_commitment_point for the remote side, we just use them as provided in each RAA. I think a better wording may be "i.e. every time we sign a new commitment transaction, we need a new per_commitment_point which are provided one at a time in RAA", or so. Note that the per_commitment_secret provided in each RAA is the secret for N-1 and the per_commiment_point is N+1 (leaving N of the commitment_signed that the RAA is "ACKing" in the middle).